app/models/exercise.rb

```diff
@@ -6,17 +6,63 @@ def self.create_exercise(blueprint_id, workout_id, sets, user_id)
     @blueprint = Blueprint.find(blueprint_id)
     exercises = @blueprint.exercises.where(workout_id: nil)
     exercises.each do |exercise|
+      weight, reps =
+        exercise.set_weight_and_reps(user_id).values_at(:weight, :reps)
       new_exercise =
         Exercise.create!(
           name: exercise["name"],
           instructions: exercise["instructions"],
           blueprint_id: exercise["blueprint_id"],
-          pic_url: exercise["pic_url"],
-          workout_id: workout_id
+          workout_id: workout_id,
+          weight: weight,
+          reps: reps,
+          user_id: user_id
         )
       sets.times do
-        ExerciseSet.create_exercise_sets new_exercise["id"], user_id
+        ExerciseSet.create_exercise_sets new_exercise["id"],
+                                         user_id,
+                                         weight,
+                                         reps
       end
     end
   end
+
+  def set_weight_and_reps(user_id)
+    @weight = 5
+    @reps = 8
+    exercise =
+      Exercise.where(name: name, user_id: user_id).order(:created_at).last
+    if exercise
```

Comment:
This is a style preference of mine, take this with grains of salt:

I prefer to avoid big chunky `if` statements due to everything getting all indent-ey and hard to read. For something like this, I'd write the logic:

```ruby
    @weight = 5
    @reps = 8
    exercise =
      Exercise.where(name: name, user_id: user_id).order(:created_at).last

    if !exercise
        return { weight: @weight, reps: @reps }
    end

   # rest of stuff
```